js/src/forum/components/SubscriptionMenu.js

```diff
@@ -59,41 +61,40 @@ export default class SubscriptionMenu extends Dropdown {
       ? 'flarum-subscriptions.forum.sub_controls.notify_email_tooltip'
       : 'flarum-subscriptions.forum.sub_controls.notify_alert_tooltip'));
 
-    const buttonProps = {
+    const buttonAttrs = {
       className: 'Button SubscriptionMenu-button ' + buttonClass,
       icon: buttonIcon,
-      children: buttonLabel,
       onclick: this.saveSubscription.bind(this, discussion, ['follow', 'ignore'].indexOf(subscription) !== -1 ? false : 'follow'),
       title: title
     };
 
     if ((notifyEmail || notifyAlert) && subscription === false) {
-      buttonProps.config = element => {
-        $(element).tooltip({
+      buttonAttrs.onupdate = vnode => {
+        $(vnode.dom).tooltip({
           container: '.SubscriptionMenu',
           placement: 'bottom',
           delay: 250,
           title
         });
       }
     } else {
-      buttonProps.config = element => $(element).tooltip('destroy');
+      buttonAttrs.onupdate = vnode => $(vnode.dom).tooltip('destroy');
     }
 
     return (
       <div className="Dropdown ButtonGroup SubscriptionMenu">
-        {Button.component(buttonProps)}
+        {Button.component(buttonAttrs, buttonLabel)}
 
         <button className={'Dropdown-toggle Button Button--icon ' + buttonClass} data-toggle="dropdown">
           {icon('fas fa-caret-down', {className: 'Button-icon'})}
         </button>
 
         <ul className="Dropdown-menu dropdown-menu Dropdown-menu--right">
-          {this.options.map(props => {
-            props.onclick = this.saveSubscription.bind(this, discussion, props.subscription);
-            props.active = subscription === props.subscription;
+          {this.options.map(attrs => {
+            attrs.onclick = this.saveSubscription.bind(this, discussion, attrs.subscription);
+            attrs.active = subscription === attrs.subscription;
 
-            return <li>{SubscriptionMenuItem.component(props)}</li>;
+            return <li>{SubscriptionMenuItem.component(attrs)}</li>;
```

Comment:
Wouldn't it be prettier to use object spread here in order to not modify the original object?

```suggestion
            return <li>{SubscriptionMenuItem.component({
                ...attrs,
                onclick: this.saveSubscription.bind(this, discussion, attrs.subscription),
                active: subscription === attrs.subscription
            })}</li>;
```